ee/server/NetworkBroker.ts

```diff
@@ -0,0 +1,225 @@
+import { ServiceBroker, Context, ServiceSchema } from 'moleculer';
+
+import { asyncLocalStorage } from '../../server/sdk';
+import { api } from '../../server/sdk/api';
+import { IBroker, IBrokerNode, IServiceMetrics } from '../../server/sdk/types/IBroker';
+import { ServiceClass } from '../../server/sdk/types/ServiceClass';
+import { EventSignatures } from '../../server/sdk/lib/Events';
+
+const events: { [k: string]: string } = {
+	onNodeConnected: '$node.connected',
+	onNodeUpdated: '$node.updated',
+	onNodeDisconnected: '$node.disconnected',
+};
+
+const lifecycle: { [k: string]: string } = {
+	created: 'created',
+	started: 'started',
+	stopped: 'stopped',
+};
+
+const {
+	INTERNAL_SERVICES_ONLY = 'false',
+	// SERVICES_ALLOWED = '',
+	WAIT_FOR_SERVICES_TIMEOUT = '10000', // 10 seconds
+	WAIT_FOR_SERVICES_WHITELIST_TIMEOUT = '600000', // 10 minutes
+} = process.env;
+
+const waitForServicesTimeout = parseInt(WAIT_FOR_SERVICES_TIMEOUT, 10) || 10000;
+const waitForServicesWhitelistTimeout = parseInt(WAIT_FOR_SERVICES_WHITELIST_TIMEOUT, 10) || 600000;
+
+export class NetworkBroker implements IBroker {
+	private broker: ServiceBroker;
+
+	private started: Promise<void>;
+
+	private whitelist = {
+		events: ['license.module', 'watch.settings'],
+		actions: ['license.hasLicense'],
+	};
+
+	// whether only internal services are allowed to be registered
+	private internalOnly = ['true', 'yes'].includes(INTERNAL_SERVICES_ONLY.toLowerCase());
+
+	metrics: IServiceMetrics;
+
+	constructor(broker: ServiceBroker) {
+		this.broker = broker;
+
+		api.setBroker(this);
+
+		this.metrics = broker.metrics;
+
+		this.started = this.broker.start();
+	}
+
+	isWhitelisted(list: string[], item: string): boolean {
+		return list.includes(item);
+	}
+
+	isActionWhitelisted(method: string): boolean {
+		return this.isWhitelisted(this.whitelist.actions, method);
+	}
+
+	isEventWhitelisted(event: string): boolean {
+		return this.isWhitelisted(this.whitelist.events, event);
+	}
+
+	async call(method: string, data: any): Promise<any> {
+		await this.started;
+
+		const context = asyncLocalStorage.getStore();
+
+		if (context?.ctx?.call) {
+			return context.ctx.call(method, data);
+		}
+
+		const services: { name: string }[] = await this.broker.call('$node.services', {
+			onlyAvailable: true,
+		});
+		if (!services.find((service) => service.name === method.split('.')[0])) {
+			return new Error('method-not-available');
+		}
+		return this.broker.call(method, data);
+	}
+
+	async waitAndCall(method: string, data: any): Promise<any> {
+		await this.started;
+
+		try {
+			await this.broker.waitForServices(
+				method.split('.')[0],
+				this.isActionWhitelisted(method) ? waitForServicesWhitelistTimeout : waitForServicesTimeout,
+			);
+		} catch (err) {
+			console.error(err);
+		}
+
+		return this.call(method, data);
```

Comment:
I noticed you're not using the `context` stored on `asyncLocalStorage` here anymore, any reason for that?